server/src/WebSocketEvents.js

```diff
@@ -763,6 +763,20 @@ eventResponses.start_dungeon = (clientSocket, data) => {
     dungeonPortal.enter(player);
 };
 
+/**
+ * @param {*} clientSocket
+ * @param {String} data.dungeonID - The ID of a dungeon manager.
+ */
+ eventResponses.leave_dungeon = (clientSocket) => {
+    if (clientSocket.inGame === false) return;
+
+    /** @type {Dungeon} */
+    const dungeon = clientSocket.entity.board.dungeon;
+    if (dungeon === undefined || dungeon === null) return;
```

Comment:
How does this become undefined?

Reply:
if for some reason entity or board weren't defined in clientSocket, then board would also be undefined. It might be possible that the "entity" in the clientSocket isn't always a player with an associated board? Anyhow, the `dungeon === undefined` part is used elsewhere in this file, so I reused the logic. I had to add the extra null check for when the player board doesn't have an associated dungeon (ie when they're in the overworld; however this function shouldn't be called from the overworld anyways)